packages/@o3r/components/builders/metadata-check/index.it.spec.ts

```diff
@@ -0,0 +1,407 @@
+  library,
+  name,
+  properties: propertiesName.map(createProp),
+  type: 'EXPOSED_COMPONENT',
+  path: ''
+});
+
+const previousConfigurationMetadata: ComponentConfigOutput[] = [
+  createConfig('@o3r/lib0', 'MyConfig0', ['prop0']),
+  createConfig('@o3r/lib1', 'MyConfig1', ['prop1']),
+  createConfig('@o3r/lib2', 'MyConfig2', ['prop2']),
+  createConfig('@o3r/lib3', 'MyConfig3', ['prop3']),
+  createConfig('@o3r/lib4', 'MyConfig4', ['prop4']),
+  createConfig('@o3r/lib5', 'MyConfig5', ['prop5']),
+  createConfig('@o3r/lib6', 'MyConfig6', ['prop6']),
+  createConfig('@o3r/lib7', 'MyConfig7', ['prop7']),
+  createConfig('@o3r/lib8', 'MyConfig8', ['prop8']),
+  createConfig('@o3r/lib9', 'MyConfig9', ['prop9'])
+];
+
+const newConfigurationMetadata = [
+  createConfig('@o3r/lib0', 'MyConfig0', ['prop0']),
+  createConfig('@o3r/lib1', 'MyConfig1', ['newProp1Name']),
+  createConfig('@o3r/lib2', 'NewConfig2', ['prop2']),
+  createConfig('@o3r/lib3', 'NewConfig3', ['newProp3Name']),
+  createConfig('@new/lib4', 'MyConfig4', ['prop4']),
+  createConfig('@new/lib5', 'MyConfig5', ['newProp5Name']),
+  createConfig('@new/lib6', 'NewConfig6', ['newProp6Name']),
+  createConfig('@o3r/lib7', 'NewConfig7', ['prop7']),
+  createConfig('@new/lib8', 'MyConfig8', ['prop8']),
+  createConfig('@new/lib9', 'MyConfig9', ['prop9'])
+];
+
+function writeFileAsJSON(path: string, content: object): void {
+  writeFileSync(path, JSON.stringify(content), { encoding: 'utf8' });
+}
+
+function getLatestPackageVersion(packageRef: string, execOptions: ExecSyncOptions): string | undefined {
+  try {
+    const result = packageManagerInfo(packageRef, ['--json'], execOptions);
+    const latestVersion = JSON.parse(result)['dist-tags']?.latest;
+    return latestVersion && coerce(latestVersion).toString() || undefined;
+  } catch (error) {
+    return undefined;
+  }
+}
+
+const initTest = (
+  builderOptions: ConfigMigrationMetadataCheckBuilderSchema,
+  newMetadata: ComponentConfigOutput[],
+  migrationData: MigrationFile<MigrationConfigData>,
+  packageNameSuffix: string
+) => {
+
+  const { workspacePath, projectName, projectPath, o3rVersion } = o3rEnvironment.testEnvironment;
+  const execAppOptions = { ...getDefaultExecSyncOptions(), cwd: projectPath };
+  const execAppOptionsWorkspace = { ...getDefaultExecSyncOptions(), cwd: workspacePath };
+  packageManagerAdd(`@o3r/components@${o3rVersion}`, execAppOptionsWorkspace);
+  packageManagerAdd(`@o3r/extractors@${o3rVersion}`, execAppOptionsWorkspace);
+  const packageJsonPath = join(projectPath, 'package.json');
+  const angularJsonPath = join(workspacePath, 'angular.json');
+  const metadataPath = join(projectPath, metadataFileName);
+  const migrationDataPath = join(projectPath, migrationDataFileName);
+
+  // Add builder options
+  const angularJson = existsSync(angularJsonPath) ? JSON.parse(readFileSync(angularJsonPath, { encoding: 'utf8' }).toString()) : undefined;
```

Comment:
I don't think this check is relevant
According to the following lines, `angularJson` is not expected to be possibly undefined